### PR TITLE
Ton2

### PR DESCRIPTION
**Pull request by enriquesouza**
The transactions done to test it are in here

https://tonscan.org/address/UQA_fNiw1Jrk-TGK2Xknb5_rPTzZGhWPVKcR8ORbNcyTKXEF

Txs
https://tonscan.org/tx/xMjpO5Qwf5xBmh0tLCcRm3vsYdUIzOyPsABuVeUpgT0=
https://tonscan.org/tx/X0bXUrC+sFgB2sO4a1e6plWm5D8qW8E7l50zrZXYB7E=
https://tonscan.org/tx/k94RsssejxvztBak6hfSOJ0qBxb3ctFuQjG2AoUR/AY=
https://tonscan.org/tx/2TbLtROpZHTIbYzkBgOhcaKEZi/mNipo/wP3xHL0o4A=
https://tonscan.org/tx/DtcRiiSV/KEQyIBsMOZwH7YRCkX4ONy+s1QmDnL+3k4=
https://tonscan.org/tx/0ed7118a2495fca110c8806c30e6701fb6110a45f838dcbeb354260e72fede4e

### CODE REVIEW

File: VultisigApp/VultisigApp/Chains/Ton.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+enum TonHelper {
+    
+    static func getPreSignedInputData(keysignPayload: KeysignPayload) throws -> Data {
+        
+        guard keysignPayload.coin.chain.ticker == "TON" else {
+            throw HelperError.runtimeError("coin is not TON")
+        }
+        
+        guard case .Ton(let sequenceNumber, let expireAt, let bounceable) = keysignPayload.chainSpecific else {
+            throw HelperError.runtimeError("fail to get to address")
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Chains/Ton.swift:22]: change the error message to "fail to get Ton chain specific"

File: VultisigApp/VultisigApp/Services/Ton/TonService.swift [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+class TonService {
+    
+    static let shared = TonService()
+
+    func broadcastTransaction(_ obj: String) async throws -> String {
+                
+        let body: [String: Any] = ["boc": obj]
+        let dataPayload = try JSONSerialization.data(withJSONObject: body, options: [])
+        let data = try await Utils.asyncPostRequest(urlString: "https://toncenter.com/api/v2/sendBocReturnHash", headers: [:], body: dataPayload)
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Services/Ton/TonService.swift:65]: can you please move this URL to `Endpoint` as well

Comment (enriquesouza) [reply VultisigApp/VultisigApp/Services/Ton/TonService.swift:65]: Oh, I forgot to push it.

File: VultisigApp/VultisigApp/Chains/Ton.swift (hunk 1 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Chains/Ton.swift]: Changed

**Review by johnnyluo — APPROVED** [review]
LGTM

**Pull request merged**